app/routes/__init__.py

```diff
@@ -137,6 +142,45 @@ async def raster_tile_cache_version_dependency(
     return dataset, version
 
 
+async def cog_asset_dependency(
+    request: Request,
+    dataset: Optional[COGDatasets] = Query(None, description=COGDatasets.__doc__),  # type: ignore
+    version: Optional[str] = Query(
+        None, description="Data API dataset version.", regex=VERSION_REGEX_NO_LATEST
+    ),
+    url: Optional[str] = Query(
+        None,
+        description="Dataset path. This needs to be set if `dataset` and `version` query parameters for a Data API dataset are not set.",
+    ),
+) -> Optional[str]:
+
+    if dataset is None and version is None and url is None:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            "Need to pass either `url` or `dataset` and `version` pair for Data API dataset.",
+        )
+
+    if dataset and version and url:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            "Need to pass either `url` or `dataset` and `version` pair, not both.",
+        )
+
+    if dataset and version:
+        folder: str = (
+            f"s3://{DATA_LAKE_BUCKET}/{dataset}/{version}/raster/epsg-4326/cog"
+        )
+        if "bands" in request.query_params:
+            return folder
```

Comment:
just flagging this, is this a workaround for when I was outputting the COG to the wrong place? Also, what about implementation? Currently saving COGs to {dataset}/{version}/raster/espg-4326/cog/{implementation}.tif"

Reply:
This is actually [multiband](https://one.newrelic.com/nr1-core/logger/logs-summary/NDA2Njk2N3xBUE18QVBQTElDQVRJT058MTQ5NjY3MjM0MA?account=4066967&begin=1719346182009&end=1719346245667&state=856b196b-826d-74a9-29ce-789d15ce6b1b) feature. The user passes the bands (implementations) so just need to specify the containing folder here.